src/state/InitState.ts

```diff
@@ -66,6 +68,20 @@ export class InitState implements IState{
         // Check if docker is running and it's the correct version
         const isCorrectDockerComposeVersion = await this.dockerService.isCorrectDockerComposeVersion();
         const isDockerStarted = await this.dockerService.checkDocker();
+        this.dockerService.isPortInUse(this.portsToCheck)
```

Comment:
Second note. Just a suggestion.
Maybe extract the `function call` and the `then` handlers to a new function. 
The goal is to have only single execution calls in onStart. 
Keeps the context clear.